platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,201 @@
+
+The validation of the static content coming from the CMS will occur during the Metalsmith build.
+
+## Specifics
+
+### Detailed Design
+
+The content validation will happen in a daily scheduled job.
+
+### Code Location
+
+The content validation script and scheduled job config will live in the [new `content-build` repo](https://github.com/department-of-veterans-affairs/content-build).
+
+### Testing Plan
+
+1. Configure a scheduled job to run a script once per workday
+2. Write an initial script that logs when the script is run
+3. Once the scheduled job is running at the correct time, replace the initial script with the actual script that runs the content validation
+
+### Logging
+
+Currently, invalid content is logged during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+### Debugging
+
+##### Broken links
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html) to have a broken link
+3. Run the build again to confirm that it is failing
+
+##### Accessibility errors
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html) to have an accessibility error
+3. Run the build again to confirm that it is failing
+
+### Caveats
+
+- Our [function to determine if a link is broken](https://github.com/department-of-veterans-affairs/vets-website/blob/e8b68850fc83fdae4386fe4ab392770dde38faee/src/site/stages/build/plugins/check-broken-links/helpers/isBrokenLink.js#L16) doesn't check if external links are broken.
+- Our [accessibility check](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/assets/js/execute-axe-check.js#L8) only renders a banner on the preview server.
+
+### Security Concerns
+
+There are no new security concerns with a separated build process.
+
+### Privacy Concerns
+
+There are no new privacy concerns with a separated build process.
+
+### Open Questions and Risks
+
+#### Risks
+
+- Neither the current content validation nor the proposed content validation will **prevent** invalid content from making it to production. But this proposal would prevent unrelated teams from being blocked by invalid content.
+
+#### Questions
+
+### Work Estimates
+
+The following estimates vary greatly depending on who's doing the work.
+
+1. Finish extracting the `content-build` repo out of the `vets-website` repo - ?
```

Comment:
I'm assuming the content-build repo will need to be sibling to vets-website at all times, correct? CMS content is everywhere, from the mega menu to user flows. Does this mean vets-website, content-build, and vagov-content will all need to be cloned side by side?

Reply:
Yes, that's correct.